cl/corpus_importer/management/commands/make_aws_manifest_files.py

```diff
@@ -263,127 +621,36 @@ def add_arguments(self, parser: CommandParser):
-                            if len(row) > 1
-                            else f"{row[0][0]}"
-                        )
-                    query_dict = {
-                        "bucket": bucket_name,
-                        "file_name": content,
-                    }
-                    writer.writerow(query_dict)
-
-                s3_client.put_object(
-                    Key=f"{file_name}_{counter}.csv",
-                    Bucket=bucket_name,
-                    Body=csvfile.getvalue().encode("utf-8"),
-                )
-
-            if options["random_sample_percentage"]:
-                # Due to the non-deterministic nature of random sampling,
-                # storing data to recover the query for future executions
-                # wouldn't be meaningful. Random queries are unlikely to
-                # produce the same results on subsequent runs.
-                logger.info(f"Finished processing {record_count} records")
-                break
-
-            counter += 1
-            last_pk = rows[-1][0]
-            records_processed = int(
-                r.hget(f"{record_type}_import_status", "records_processed")
-                or 0
-            )
-
-            # Store the results of this iteration and log the progress
-            r.hset(
-                f"{record_type}_import_status",
-                mapping={
-                    "last_pk": last_pk,
-                    "next_iteration_counter": counter,
-                    "records_processed": records_processed + record_count,
-                },
-            )
-            logger.info(
-                "\rRetrieved {}/{}, ({:.0%}), last PK processed: {},".format(
-                    record_count + records_processed,
-                    total_number_of_records,
-                    (record_count + records_processed)
-                    * 1.0
-                    / total_number_of_records,
-                    last_pk,
+        monthly_export = options["monthly_export"]
+
+        if monthly_export:
+            last_export_key = f"bulk_import:{record_type}"
+            last_export_timestamp = r.get(last_export_key)
+            if not last_export_timestamp:
+                return logger.info(
+                    "Warning: No previous timestamp for bulk import found for "
+                    f"record type: {record_type}."
                 )
+            compute_monthly_export(
+                record_type,
+                datetime.fromisoformat(last_export_timestamp),
+                options,
             )
+        else:
+            export_records_in_batches(r, record_type, options)
 
-        # Removes the key from the cache after a successful execution
-        r.delete(f"{record_type}_import_status")
+        # Store the timestamp of the last successful bulk import for this
+        # record type, expiring after 45 days.
+        r.set(last_export_key, str(timezone.now()), 60 * 60 * 24 * 45)
```

Comment:
I see that `last_export_key` is only used when performing a `monthly_export`, right?

If that's the case, it should be moved inside the `if monthly_export:` block.
Otherwise, the Redis set operation will fail for non-monthly exports, since `last_export_key` is only defined for `monthly_export`.

Reply:
Thanks for catching this. 

it's important to create the timestamp after both the monthly and full exports because our Lambda functions write to the same S3 folder. This PR  aims to append newer records to our existing dataset. Additionally, storing the timestamp immediately after a full export could prove valuable in the future, particularly if we integrate new models and need to pinpoint the exact time of the full export. To achieve this, I'm moving the `last_export_key` variable outside of the if statement.